README.md

```diff
@@ -1,102 +1,105 @@
-# certbot-dns-desec
+# certbot-dns-desec: Get Let's Encrypt Certificates for Domains Hosted at deSEC
 
 ![main branch CI test status](https://github.com/desec-io/certbot-dns-desec/workflows/Tests/badge.svg?branch=main)
 [![pypi badge](https://img.shields.io/pypi/v/certbot-dns-desec.svg)](https://pypi.org/project/certbot-dns-desec/)
 
-⚠ This plugin is under development, API and CLI might change! ⚠
+Certbot plugin to obtain TLS certificates from Let's Encrypt for domains hosted with deSEC.io, using the DNS challenge
+challenge mechanism.
 
-[deSEC](https://desec.io/) DNS Authenticator plugin for [Certbot](https://certbot.eff.org/)
 
-This plugin automates the process of completing a ``dns-01`` challenge by
-creating, and subsequently removing, TXT records using the deSEC DNS API.
+## Installation
 
-## Configuration of deSEC
+To get certificates from Let's Encrypt, install certbot and this plugin.
+There are many ways to install certbot, this guide uses Python's `pip`:
 
-Log in at [deSEC](https://desec.io/) and create the domain you would like to use and a token for DNS management.
+```
+python3 -m pip install certbot certbot-dns-desec
+```
 
-## Installation
+## Prerequisites
 
-This certbot plugin can be installed using `pip`:
+To get a Let's Encrypt certificate for your domain,
+you need an access token for the deSEC.io account that holds the domain you need the certificate for.
+Also make sure that your domain name has been delegated to deSEC according to deSEC's instructions.
 
-    python3 -m pip install certbot-dns-desec
 
+## Issue Certificate
 
-## Command Line Interface
+To issue and renew certificates using `certbot-dns-desec`, an access token to your deSEC account is required.
+To store such a token in a secure location, use, e.g.:
 
-To start using DNS authentication with deSEC, pass the following arguments on
-certbot's command line:
+```
+DOMAIN=example.com
+TOKEN=your-desec-access-token
+sudo mkdir /etc/letsencrypt/secrets/
+sudo chmod 700 /etc/letsencrypt/secrets/
+echo "dns_desec_token = $TOKEN" | sudo tee /etc/letsencrypt/secrets/$DOMAIN.ini
+sudo chmod 600 /etc/letsencrypt/.secrets/$DOMAIN.ini
```

Comment:
You could consider using `umask`. Very ugly, however:
```sh
sudo -- sh -c "umask 0077 && mkdir -p etc/letsencrypt/secrets/ && echo 'dns_desec_token = $TOKEN' > etc/letsencrypt/secrets/$DOMAIN.ini"
```

Reply:
I believe the verbose version is more beginner friendly. Experts know what's the goal here and can `umask` away :)